src/main/scala/com/criteo/slab/app/StateService.scala

```diff
@@ -56,16 +56,46 @@ private[slab] class StateService(
       }
   }
 
-  // Stats of last n days
-  def stats(board: String): Future[Map[Long, Stats]] = memoize(Duration.create(12, TimeUnit.HOURS)) {
-    logger.info(s"Updating statistics of $board")
-    val now = Instant.now()
+  def stats(board: String): Future[Map[Long, Double]] = {
+    val now = ZonedDateTime.now(ZoneOffset.UTC)
     executors.find(_.board.title == board)
-      .fold(Future.failed(NotFoundError(s"$board does not exist")): Future[Map[Long, Stats]]) {
-        _
-          .fetchHistory(now.minus(statsDays, ChronoUnit.DAYS), now)
-          .map(getStatsByHour)
+      .fold(Future.failed(NotFoundError(s"$board does not exist")): Future[Map[Long, Double]]) {
+        executor =>
+
+          def getSloCacheDuration(from: ZonedDateTime): Duration =
+            if (from.plus(1, ChronoUnit.MONTHS).compareTo(now) > 0)
+            // expire cache fro current month more often
+              Duration.create(1, TimeUnit.HOURS)
+            else
+            // don't expire all caches in the same time
+              Duration.create(12, TimeUnit.HOURS).plus(Duration.create((Math.random() * 30).toInt, TimeUnit.MINUTES))
+
+          //cache only full months
+          def boardMonthlySloInner(board: String, from: ZonedDateTime, until: ZonedDateTime): Future[Map[Long, Double]] = memoize(getSloCacheDuration(from)) {
+            executor.fetchHourlySlo(from.toInstant, until.toInstant).map(_.toMap)
+          }
+
+          val from = now.minus(statsDays, ChronoUnit.DAYS)
+          val until = now
+
+          val fromStartOfMonth = from.truncatedTo(ChronoUnit.DAYS).withDayOfMonth(1)
+
+          val months = ChronoUnit.MONTHS.between(fromStartOfMonth, until).toInt
+          //take take one extra period to avoid timezone issues
```

Comment:
do we really need extra months to handle this? I guess an offset by day would cover all zones